packages/eas-cli/src/project/ios/target.ts

```diff
@@ -24,17 +24,57 @@ export async function resolveTargetsAsync(
     buildConfiguration,
   });
 
-  if (applicationTarget.dependencies && applicationTarget.dependencies.length > 0) {
-    for (const dependency of applicationTarget.dependencies) {
+  const dependencies = await resolveDependenciesAsync({
+    exp,
+    projectDir,
+    buildConfiguration,
+    target: applicationTarget,
+    bundleIdentifier,
+  });
+  if (dependencies.length > 0) {
+    result.push(...dependencies);
+  }
+
+  return result;
+}
+
+async function resolveDependenciesAsync({
+  exp,
+  projectDir,
+  buildConfiguration,
+  target,
+  bundleIdentifier,
+}: {
+  exp: ExpoConfig;
+  projectDir: string;
+  buildConfiguration?: string;
+  target: IOSConfig.Target.Target;
+  bundleIdentifier: string;
+}): Promise<Target[]> {
+  const result: Target[] = [];
+
+  if (target.dependencies && target.dependencies.length > 0) {
+    for (const dependency of target.dependencies) {
+      const dependencyBundleIdentifier = await getBundleIdentifierAsync(projectDir, exp, {
+        targetName: dependency.name,
+        buildConfiguration,
+      });
       result.push({
         targetName: dependency.name,
         buildConfiguration,
-        bundleIdentifier: await getBundleIdentifierAsync(projectDir, exp, {
-          targetName: dependency.name,
-          buildConfiguration,
-        }),
+        bundleIdentifier: dependencyBundleIdentifier,
         parentBundleIdentifier: bundleIdentifier,
       });
+      const dependencyDependencies = await resolveDependenciesAsync({
+        exp,
+        projectDir,
+        buildConfiguration,
+        target: dependency,
+        bundleIdentifier: dependencyBundleIdentifier,
```

Comment:
shouldn't parentBundleIdentifier for the dependency of a dependency point always to the aplicationTraget bundle identifier? The main thing I'm wondering here is whether the credential manager and www credentials api will handle that correctly and if we would flatten that structure we wouldn't need to worry about it.

Reply:
I'd prefer to preserve the original hierarchy.